test/ecto_cursor_based_stream_test.exs

```diff
@@ -149,6 +149,33 @@ defmodule EctoCursorBasedStreamTest do
 
       assert options == [prefix: "public", timeout: 10_000, log: false]
     end
+
+    defmodule ParallelRepoStub do
+      use EctoCursorBasedStream
+
+      def all(_query, _options) do
+        :timer.sleep(100)
+
+        [%User{id: System.monotonic_time()}]
+      end
+    end
+
+    test "if :parellel option is given, fetches data in parallel" do
+      [not_parallel_duration, parallel_duration] =
+        for parallel? <- [false, true] do
+          :timer.tc(fn ->
+            ParallelRepoStub.cursor_based_stream(User, parallel: parallel?)
+            |> Stream.each(fn _ -> :timer.sleep(100) end)
+            |> Enum.take(5)
```

Comment:
Are we interested in the order of results?

Reply:
I wonder if we should inform the developer that using parallel may change the order.

Reply:
Order of result does not change, tasks are invoked one by one. Fetching rows is in parallel to the stream iterating over results. I could rename option if it would make it more clear 🤔

Reply:
Hmm, it's a tricky one, I have no idea how to name it properly, to be concise, and to get enough clarity.